dali/kernels/test/alloc_test.cc

```diff
@@ -25,12 +25,23 @@ TEST(KernelAlloc, AllocFree) {
   size_t size = 1<<20;  // 1 MiB
   const char *names[static_cast<int>(AllocType::Count)] = { "Host", "Pinned", "GPU", "Unified" };
   for (int i = 0; i < static_cast<int>(AllocType::Count); i++) {
-    AllocType alloc = static_cast<AllocType>(i);
-    void *mem = memory::Allocate(alloc, size);
-    EXPECT_EQ(cudaGetLastError(), 0) << "Error when allocating for " << names[i];
-    ASSERT_NE(mem, nullptr);
-    memory::GetDeleter(alloc)(mem);
-    EXPECT_EQ(cudaGetLastError(), 0) << "Error when freeing for " << names[i];
+    try {
+      AllocType alloc = static_cast<AllocType>(i);
+      void *mem = memory::Allocate(alloc, size);
+      EXPECT_EQ(cudaGetLastError(), 0) << "Error when allocating for " << names[i];
+      // use ThrowMemoryError to call cudaGetLastError from the kernel libray
+      // (and use CUDA rt status from there)
+      if (!mem) {
+        memory::ThrowMemoryError(alloc, size);
+      }
+      memory::GetDeleter(alloc)(mem);
+      EXPECT_EQ(cudaGetLastError(), 0) << "Error when freeing for " << names[i];
+    } catch (const CUDAError &e) {
+      // skip this case if Unified is not supported
+      bool not_supported = (e.is_drv_api() && e.drv_error() == CUDA_ERROR_NOT_SUPPORTED) ||
+                           (e.is_rt_api() && e.rt_error() == cudaErrorNotSupported);
+      EXPECT_TRUE(not_supported) << "Unexpected CUDA exception: " << e.what();
+    }
```

Comment:
There is something called `GTEST_SKIP()` and I think it might be useful here. Maybe we could consider using it as it more clearly shows the intent of skipping the test if some runtime condition is met. It could be combined with `cudaDevAttrManagedMemory` query, I think.
Regardless, I think it is ok as it is.

Reply:
We don't want to skip whole test, only a one iteration, as "Host", "Pinned", "GPU" should work.

Reply:
Right, we would have to change it to parameterized test and run them separately. Not worth to do it, I think.